code/controllers/subsystem/timer.dm

```diff
@@ -246,7 +246,7 @@ var/datum/subsystem/timer/SStimer
 	if (callBack && callBack.object && callBack.object != GLOBAL_PROC && callBack.object.active_timers)
 		callBack.object.active_timers -= src
 		UNSETEMPTY(callBack.object.active_timers)
-
+		qdel(callBack)
```

Comment:
a thing having references to other things doesn't keep it from gcing. so the only situation this would prevent a object from qdeleting is if something kept a reference to the callback, and since callback's destroy doesn't clear up any references to itself, you haven't actually prevented a harddelete, you just moved it to callback